src/pages/guides/dart/flutter.mdx

```diff
@@ -0,0 +1,1125 @@
+          return SizeTransition(
+            sizeFactor: animation,
+            child: Center(
+              child: TextButton.icon(
+                onPressed: () {
+                  favorites.togglefavorite(pair);
+                },
+                // If the word pair was favorited, show a heart next to it
+                icon: favorites.hasfavorite(pair)
+                    ? Icon(Icons.favorite, size: 12)
+                    : SizedBox(),
+                label: Text(
+                  pair.asLowerCase,
+                ),
+              ),
+            ),
+          );
+        },
+      ),
+    );
+  }
+}
+```
+
+With that built you can add it to the `GeneratorPage`.
+
+```dart {{ label: "lib/pages/generator.dart" }}
+return Center(
+  child: Column(
+    mainAxisAlignment: MainAxisAlignment.center,
+    children: [
+      Expanded( // <- allows the list to extend to the top of the page
+        flex: 3,
+        child: HistoryListView(), // <- Add the history list view here
+      ),
+      SizedBox(height: 10),
+      Card(
+        ...
+      ),
+      ...
+      Spacer(flex: 2), // <- Stops the Card being pushed to the bottom of the page
+    ]
+  )
+);
+```
+
+If you reload the flutter app it should now display your history when you click through the words.
+
+![generator page with history](/docs/images/guides/flutter/main_page_2.png)
+
+### favorites Page
+
+The favorites page will simply list all the favorites and the number that have been liked:
+
+```dart {{ label: "lib/pages/favorites.dart" }}
+import 'package:flutter/material.dart';
+import 'package:word_generator/providers/favorites.dart';
+import 'package:provider/provider.dart';
+
+class favoritesPage extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    var favorites = context.watch<favoritesProvider>();
+
+    // If the favorites list is still loading then show a spinning circle.
+    if (favorites.isLoading) {
+      return Center(
+          child: SizedBox(
+        width: 40,
+        height: 40,
+        child: CircularProgressIndicator(color: Colors.blue),
+      ));
+    }
+
+    // Otherwise return a list of all the favorites
+    return ListView(
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(20),
+          // Display how many favorites there are
+          child: Text('You have '
+              '${favorites.favorites.length} favorites:'),
+        ),
+        // Create a list tile for every favorite in the list of favorites
+        for (var favorite in favorites.favorites)
+          ListTile(
+            leading: Icon(Icons.favorite), // <- A heart icon
+            title: Text(favorite.name),
+          ),
+      ],
+    );
+  }
+}
+```
+
+You might notice at no point is the favorites list actually being fetched, so the `favoritesProvider` will always contain an empty favorites list. This will be handled in the next section where we build the navigation.
```

Comment:
```suggestion
You might notice that at no point is the favorites list actually being fetched, so the `favoritesProvider` will always contain an empty favorites list. This will be handled in the next section where we build the navigation.
```